pkg/cloudmeta/metadata_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2017 ScyllaDB
+
+package cloudmeta
+
+import (
+	"context"
+	"fmt"
+	"testing"
+)
+
+func TestGetInstanceMetadata(t *testing.T) {
+	t.Run("when there is no active providers", func(t *testing.T) {
+		cloudmeta := &CloudMeta{}
+
+		meta, err := cloudmeta.GetInstanceMetadata(context.Background())
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+		if meta.InstanceType != "" {
+			t.Fatalf("meta.InstanceType should be empty, got %v", meta.InstanceType)
+		}
+
+		if meta.CloudProvider != "" {
+			t.Fatalf("meta.CloudProvider should be empty, got %v", meta.CloudProvider)
+		}
+	})
```

Comment:
Cool that you are writing detailed tests!

The lack of empty line between 18-19 and an empty line 22 seems inconsistent?

I would say that it's safer and more intuitive to return an error when no provider is specified and the instance metadata is queried. It most likely indicates an error in code development. We can make it a named error so that it can be handled accordingly if needed.
What is you opinion on that?

Reply:
yep, agree, it's better to return error!